bin/meshroom_batch

```diff
@@ -156,60 +177,30 @@ with meshroom.core.graph.GraphModification(graph):
-                # and re-join the path together.
-                elif uniqueInitNode:
-                    nodeName = uniqueInitNode.getName()
-                    inputGroup = [":".join(inputGroup)]
-                else:
-                    print('Syntax error in input argument')
-            # If a node name has been provided and the platform is Windows, the length of inputGroup might be
-            # 3: first the node name, then the first path of the input path, then the rest of the input path.
-            # The input path should also be re-joined here.
-            elif len(inputGroup) == 3:
-                nodeName = inputGroup[0]
-                inputGroup[1] = ":".join(inputGroup[1:])
-                inputGroup.pop(-1)
-            else:
-                print('Syntax error in input argument')
-                sys.exit(1)
-            nodeInputs = inputGroup[-1].split(';')
-            mapInputs[nodeName] = [os.path.abspath(path) for path in nodeInputs]
-        return mapInputs
-
-    # get init nodes
-    initNodes = graph.findInitNodes()
-    uniqueInitNode = initNodes[0] if (len(initNodes) == 1) else None
-
-    # parse inputs for each init node
-    mapInput = parseInputs(args.input, uniqueInitNode)
-
-    # parse recursive inputs for each init node
-    mapInputRecursive = parseInputs(args.inputRecursive, uniqueInitNode)
-
-    # feed inputs and recursive inputs to init nodes
-    for initNode in initNodes:
-        nodeName = initNode.getName()
-        inp = mapInput.get(nodeName)
-        inpRec = mapInputRecursive.get(nodeName)
-        if not inp and not inpRec:
-            print(f'Nothing to compute for node {nodeName}.')
-            sys.exit(1)
-        initNode.nodeDesc.initialize(initNode, inp, inpRec)
+    if args.input:
+        # get init nodes
+        initNodes = graph.findInitNodes()
+        initNodesNames = [n.getName() for n in initNodes]
+
+        # parse inputs for each init node
+        mapInput = parseInitInputs(args.input)
+
+        # parse recursive inputs for each init node
+        mapInputRecursive = parseInitInputs(args.inputRecursive)
+
+        # check that input nodes exist in the pipeline template
+        for nodeName in mapInput.keys() | mapInputRecursive.keys():
+            if nodeName and nodeName not in initNodesNames:
+                raise RuntimeError(f"Failed to find the Init Node '{nodeName}' in your pipeline.\nAvailable Init Nodes: {initNodesNames}")
+
+        # feed inputs (recursive and non-recursive paths) to corresponding init nodes
+        for initNode in initNodes:
+            nodeName = initNode.getName()
+            # Retrieve input per node and inputs for all init node types
+            input = mapInput.get(nodeName, []) + mapInput.get("", [])
+            # Retrieve recursive inputs
+            inputRec = mapInputRecursive.get(nodeName, []) + mapInputRecursive.get("", [])
```

Comment:
There is kind of a corner case when some of the `InitNodes` are not set. By calling `initialize` with `[]` values, we may run into errors within the `initialize` function that are not explicit (for example, with CameraInit nodes, we end up with this error: `ValueError("No valid input file or no image in the provided directories")`.

I think it'd be best to either raise an exception before calling `initialize` or to set the node with the value used to set the previous one (similarly to what is done when no specific instance of an `InitNode` is specified in the command line) when we get the default `[]` value.

Reply:
Done